pkg/cmd/taskrun/delete.go

```diff
@@ -359,3 +384,28 @@ func keepTaskRunsByAgeAndNumber(taskRuns *v1beta1.TaskRunList, since int, keep i
 	}
 	return todelete, tokeep
 }
+
+func trsWithOwnerPrFinished(cs *cli.Clients, runs *v1beta1.TaskRunList) *v1beta1.TaskRunList {
+	var filteredTrs v1beta1.TaskRunList
+	for _, tr := range runs.Items {
+		if ownerPrFinished(cs, tr) {
+			filteredTrs.Items = append(filteredTrs.Items, tr)
+		}
+	}
+	return &filteredTrs
+}
+
+func ownerPrFinished(cs *cli.Clients, tr v1beta1.TaskRun) bool {
+	for _, ref := range tr.GetOwnerReferences() {
+		if ref.Kind == pipeline.PipelineRunControllerName {
+			pr, err := pipelinerunpkg.Get(cs, ref.Name, metav1.GetOptions{}, tr.Namespace)
+			if err != nil {
+				return false
+			}
+			for _, c := range pr.Status.Conditions {
+				return c.Reason != "Running"
```

Comment:
there can be case that pipeline is still pending etc.

Reply:
:+1:  here I will add similar logic of completion time as we did for running taskruns